cpp/src/parquet/statistics_test.cc

```diff
@@ -111,21 +117,28 @@ TEST(Comparison, UnsignedByteArray) {
 }
 
 TEST(Comparison, SignedFLBA) {
-  int size = 10;
+  int size = 4;
   auto comparator =
       MakeComparator<FLBAType>(Type::FIXED_LEN_BYTE_ARRAY, SortOrder::SIGNED, size);
 
-  std::string s1 = "Anti123456";
-  std::string s2 = "Bunkd123456";
-  FLBA s1flba = FLBAFromString(s1);
-  FLBA s2flba = FLBAFromString(s2);
-  ASSERT_TRUE(comparator->Compare(s1flba, s2flba));
+  std::vector<uint8_t> byte_values[] = {
+      {0x80, 0, 0, 0},          {0xFF, 0xFF, 0x01, 0},    {0xFF, 0xFF, 0x80, 0},
+      {0xFF, 0xFF, 0xFF, 0x80}, {0xFF, 0xFF, 0xFF, 0xFF}, {0, 0, 0x01, 0x01},
+      {0, 0x01, 0x01, 0},       {0x01, 0x01, 0, 0}};
+  std::vector<FLBA> values_to_compare;
+  for (auto& bytes : byte_values) {
+    values_to_compare.emplace_back(FLBA(bytes.data()));
+  }
 
-  s1 = "Bünk123456";
-  s2 = "Bunk123456";
-  s1flba = FLBAFromString(s1);
-  s2flba = FLBAFromString(s2);
-  ASSERT_TRUE(comparator->Compare(s1flba, s2flba));
+  for (size_t x = 0; x < values_to_compare.size(); x++) {
+    EXPECT_FALSE(comparator->Compare(values_to_compare[x], values_to_compare[x])) << x;
+    for (size_t y = x + 1; y < values_to_compare.size(); y++) {
+      EXPECT_TRUE(comparator->Compare(values_to_compare[x], values_to_compare[y]))
+          << x << " " << y;
+      EXPECT_FALSE(comparator->Compare(values_to_compare[y], values_to_compare[x]))
+          << y << " " << x;
+    }
+  }
```

Comment:
Do these tests cover the decimal case?

Reply:
I think for FLBA these tests should be fairly complete.  the bytes can be interepreted as big-endian two's complement number.   I think I need to add an additional case that covers sign extension equality from the example I gave above.  Is your concern upon representation here?  Or would you like me to explicitly use util::Decimal* classes for the test cases?

Reply:
I was merely wondering if we're sure that decimal columns trigger the code paths under test. But I'll trust you that it's the case :-)

Reply:
you probably shouldn't trust me.  I added a test when writing a column of Decimal128 values to arrow_reader_writer_test which failed without this change.  This required fixing some cases with extracting scalar values and there are still some broken aspects to the extraction.  I'll open up some JIRAs to fix these:
1. Decimals encoded as Ints aren't extracted correctly.
2. The decimal type chosen might not correspond to the decoded decimal type because we don't pass the arrow schema information through to the decoding. (i.e. we depend solely on precision for deciding Decimal128Scalar or Decimal256Scalar).